### PR TITLE
Overwatch: Laser Battery Orbital Support Option

### PR DESCRIPTION
**Pull request by Surrealaser**
:cl: by Surrealistik
add: Orbital laser battery orbital fire support added. Has a shared cooldown with the rail gun. Ignites xenos and starts fires instead of staggering/slowing/knocking down. Laser is powered by the ship weapon APC and draws on the power grid instead of using limited ammo.
/:cl:

### CODE REVIEW

**Comment by Alterist** [comment]
You changed the railgun from structure to machinery, but did you change the map?

File: code/modules/cm_marines/orbital_cannon.dm [hunk 1]
```diff
@@ -562,8 +569,26 @@ var/obj/structure/ship_rail_gun/almayer_rail_gun
...
-/obj/structure/ship_rail_gun/ex_act()
+/obj/machinery/ship_rail_gun/ex_act()
 	return
 
-/obj/structure/ship_rail_gun/bullet_act()
+/obj/machinery/ship_rail_gun/bullet_act()
 	return
+
+/obj/machinery/ship_rail_gun/proc/fire_laser_battery(turf/T, mob/user)
+	set waitfor = 0
```
Comment (DominikPanic) [inline code/modules/cm_marines/orbital_cannon.dm]:
```suggestion
	set waitfor = FALSE
```

File: code/modules/cm_marines/orbital_cannon.dm [hunk 2]
```diff
@@ -562,8 +569,26 @@ var/obj/structure/ship_rail_gun/almayer_rail_gun
...
+		to_chat(user, "<span class='warning'>[src]'s APC has run out of charge.</span>")
+		return
+	machine_current_charge -= ORBITAL_LASER_COST
+	flick("Railgun_firing",src)
+	cannon_busy = TRUE
+	last_firing = world.time
+	playsound(loc, 'sound/weapons/tank_smokelauncher_fire.ogg', 70, 1)
+	playsound(loc, 'sound/weapons/pred_plasma_shot.ogg', 70, 1)
+	var/turf/target = locate(T.x + pick(-2,2), T.y + pick(-2,2), T.z)
+	sleep(15)
```
Comment (DominikPanic) [inline code/modules/cm_marines/orbital_cannon.dm]: This should be a callback anyway.

File: code/modules/cm_marines/orbital_cannon.dm [hunk 3]
```diff
@@ -533,19 +533,26 @@ var/obj/structure/ship_rail_gun/almayer_rail_gun
 	bound_height = 64
 	bound_y = 64
 	unacidable = TRUE
+	use_power = 1
```
Comment (DominikPanic) [inline code/modules/cm_marines/orbital_cannon.dm]:
```suggestion
	use_power = TRUE
```

File: code/modules/cm_marines/orbital_cannon.dm [hunk 4]
```diff
@@ -562,8 +569,26 @@ var/obj/structure/ship_rail_gun/almayer_rail_gun
...
-/obj/structure/ship_rail_gun/bullet_act()
+/obj/machinery/ship_rail_gun/bullet_act()
 	return
+
+/obj/machinery/ship_rail_gun/proc/fire_laser_battery(turf/T, mob/user)
+	set waitfor = 0
+	if(cannon_busy)
+		return
+	if(ORBITAL_LASER_COST > machine_current_charge)
+		to_chat(user, "<span class='warning'>[src]'s APC has run out of charge.</span>")
```
Comment (Menshin) [inline code/modules/cm_marines/orbital_cannon.dm]: `machine_current_charge` is the `obj` internal battery, not the APC cell.

**Comment by Alterist** [comment]
Instead of making it a part of the railgun, you could make it a new cannon, using the gauss_cannon sprite found in the same dmi.

**Comment by Borisvanmemes** [comment]
How big is the area of effect?

File: code/modules/cm_marines/orbital_cannon.dm (hunk 3 above)
Comment (Menshin) [reply code/modules/cm_marines/orbital_cannon.dm]:
```suggestion
	use_power = IDLE_POWER_USE
```

File: code/modules/cm_marines/orbital_cannon.dm (hunk 4 above)
Comment (Surrealaser) [reply code/modules/cm_marines/orbital_cannon.dm]: Was from an earlier implementation; forgot to change it

**Comment by Menshin** [comment]
Laser battery `detonate()` is picking random turfs in `range(3, impact)` so, in a square of turfs within 3 tiles of the impact turf.

File: code/modules/cm_marines/orbital_cannon.dm [hunk 5]
```diff
@@ -533,7 +533,7 @@ var/obj/machinery/ship_rail_gun/almayer_rail_gun
 	bound_height = 64
 	bound_y = 64
 	unacidable = TRUE
-	use_power = 1
+	use_power = TRUE
```
Comment (Menshin) [inline code/modules/cm_marines/orbital_cannon.dm]: `use_power` should rather be `IDLE_POWER_USE` than `TRUE`, as there's three possible states.

**Comment by spookydonut** [comment]
map edit requires a mapmerge

**Comment by AdamElTablawy** [comment]
this will be bad with such a small range

**Comment by DominikPanic** [comment]
Stale

**Pull request closed without merging**